src/utils.ts

```diff
@@ -1726,6 +1984,74 @@ export class Triangle extends AbstractTriangle {
   }
 }
 
+
+export class Contour extends newShape {
+  constructor(
+    public lines: (Arc | LineSegment)[] = [],
+    public isFilled: boolean = false
+  ) {
+    super();
+    this.buildPath();
+  }
+
+  public static deserialize(data: any, scale: Vertex): Contour {
+    const lines = [];
+    data.plot_data_primitives.forEach(primitive => {
+      if (primitive.type_ == "linesegment2d") lines.push(LineSegment.deserialize(primitive, scale));
```

Comment:
Since `linesegment2` and `arc` are newShapes, can't we directly deserialize in a map in a more generic way ? Like this : 

```js
const lines = data.plot_data_primitives.map(newShape.deserialize(primitive, scale))
```

with the deserialize method of newShape being a factory like you did with the vertex one ?

Or is that we want to filter the list by only including these two ? In which case we could filter beforehand and then map

Reply:
Thanks, refactored and changed for a map